### ISSUE TITLE
Set Dislocation Ring for selective attraction

### ISSUE THREAD

**Issue by Tillllt** [Extegral/Enigmatic-Legacy#49]
**Provide a detailed and clear description of feature or change that you want to offer.**
Just possibility to set DR to not attract some items. Like it works with Keystone of the Oblivion.
**Describe why you want this. If there's a reason.**
That feeling when you return to your base, forget to take off the DR and take all stuff (bamboo, sugar cane, etc.) from your farms. You actually need that items in your stashes not in inventory. But that option can solve the problem of. From an aesthetic point of view it will be reasonable to put that option for Dislocation Ring only and not for Magnetic Ring, because DR has eye and can distinguish. Also we have farms in late survival when most likely we already have DR instead MR. Still MR can't attract items through blocks.

**Comment by Extegral** [comment]
Fairly interesting idea. Inconvenience of uncontrollably picking up stuff in unwanted places is usually fixed through other means, like some kind of block that does suppress effects of magnet rings in an area around itself, which would probably be a better solution for outlined scenario with farms. [Demagnetize](https://www.curseforge.com/minecraft/mc-mods/demagnetize) usually provided a good tool to take care of that, which Enigmatic Legacy's magnets were compatible with since 1.14 releases. And I wanted to say that Demagnetize itself, unfortunately, wasn't ported past 1.14 - but when checking out project page I have noticed that 1.15/1.16 ports were released just a day ago, so here's at least one possible solution.

But from my brief experience of playing a modpack with my own mod a while ago, I have concluded that having an ability to instantly collect all the items such a vast area isn't always so useful. I often ended up with a ton of garbage in my inventory that I would rarely want to pick up under normal circumstances, and getting rid of it after that was tricky without having something like a lava pool nearby, because for that I would require to either unequip the ring, or hold Shift and crouch all the way up to 16 blocks away from where I've thrown out unwanted items, which isn't exactly great gameplay experience. Addition of Keystone of The Oblivion did partially solve such problem, but it's a different tool that doesn't always perfectly help with this purpose.

Thus, after thinking for a while, I guess I'll implement some way of attaching white/blacklist functionality to Dislocation Ring. I suppose it will not be present by default, but there will be special craftable items (might literally be called "The Whitelist" and "The Blacklist"), which can be combined with ring. After that it will have item list that can be operated with in a way similar to Keystone of The Oblivion.

These lists might even have some more application to them in future. Like, for instance, it will be possible to operate with them without attaching them to an item first, they could be safely separated from item they were attached to, there will be ways to copy and combine them, and apply their effects to other items that might have something to do with them, like abovementioned keystone, or whatever might come with new updates of the mod.

**Comment by Tillllt** [comment]
Cool, thanks!

**Comment by darkarya** [comment]
Maybe you can just put a keybind to turn on/off the magnetization like the keybind to open the ender chest with the ring of ender.
It'll be more effective i think !

**Comment by Aizistral** [comment]
> Maybe you can just put a keybind to turn on/off the magnetization like the keybind to open the ender chest with the ring of ender.
> It'll be more effective i think !

Can't you just unequip the ring if you don't need it? Also, continuous sneaking does suppress ring's effects.

**Comment by darkarya** [comment]
No, the ring is very useful and it is just painful to have to remove it in specific areas like farms that require items on the ground. Being able to turn it on / off would really add value to the ring!

**Comment by Aizistral** [comment]
Alright, I will consider.

**Comment by darkarya** [comment]
Thank you a lot ! and Thank you for your work it's amazing !
